src/EditorFeatures/CSharpTest/AddBraces/AddBracesTests.cs

```diff
@@ -551,13 +1026,18 @@ public void Dispose()
     {
         throw new NotImplementedException();
     }
-}");
+}",
+                (PreferBracesPreference)bracesPreference,
+                expectDiagnostic);
         }
 
-        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAddBraces)]
-        public async Task FireForLockWithoutBraces()
+        [Theory, Trait(Traits.Feature, Traits.Features.CodeActionsAddBraces)]
+        [InlineData((int)PreferBracesPreference.None, false)]
+        [InlineData((int)PreferBracesPreference.WhenMultiline, false)]
+        [InlineData((int)PreferBracesPreference.Always, true)]
+        public async Task FireForLockWithoutBraces(int bracesPreference, bool expectDiagnostic)
         {
-            await TestInRegularAndScriptAsync(
+            await TestMissingOrInRegularAndScriptAsync(
```

Comment:
even if painful to write, i would expect tests showin what happens if the lock expr spans multi-lines, or hte lck-body spans multiple lines.

Reply:
Adding tests is no problem. I decided not to run all combinations of inputs, but can add interesting cases as we see them.

Reply:
➡️ Added tests for multiline conditions in `using` statements